src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
-// write code here
+const headerOfTable = document.querySelector('thead tr');
+const headers = headerOfTable.querySelectorAll('th');
+
+const sortTable = (index) => {
+  const rows = document.querySelectorAll('tbody tr');
+  const column = [];
+
+  rows.forEach((row) => {
+    column.push(row.cells[index].cloneNode(true));
+  });
+
+  column.sort((a, b) => {
+    const textA = a.innerText.trim();
+    const textB = b.innerText.trim();
+
+    const extractNumericValue = (text) => {
+      return parseFloat(text.replace(/[$,]/g, ''));
+    };
+
+    const valueA = extractNumericValue(textA);
+    const valueB = extractNumericValue(textB);
+
+    if (!isNaN(valueA) && !isNaN(valueB)) {
+      return valueA - valueB;
+    } else {
+      return textA.localeCompare(textB);
+    }
+  });
+
+  column.forEach((cell, i) => {
+    rows[i].replaceChild(cell, rows[i].cells[index]);
+  });
```

Comment:
The current implementation of replacing cells in the rows with sorted cells is incorrect. The `column` array contains cloned nodes, which means they are detached from the original DOM. When you replace a cell in a row with a cloned node, it doesn't update the actual table structure correctly. Instead, consider sorting the rows themselves based on the values in the `column` array and then re-adding them to the table body.